geowebcache/core/src/main/java/org/geowebcache/GeoWebCacheDispatcher.java

```diff
@@ -284,6 +288,10 @@ protected ModelAndView handleRequestInternal(HttpServletRequest request,
             OWSException owsE = (OWSException) e;
             writeFixedResponse(response, owsE.getResponseCode(), owsE.getContentType(),
                     owsE.getResponse(), CacheResult.OTHER);
+        } catch (SecurityException e) {
+            writeFixedResponse(response, 403, "text/plain",
+                    new ByteArrayResource("Not Authorized".getBytes()), CacheResult.OTHER);
```

Comment:
Do we just care about securing the layers, or also hiding their existence? I think GeoServer uses 404 for both secured layers and non-existing layers so you can't check if the layers exist based on the error response.

Reply:
This is why I added `GeoWebCacheException` as an option to be thrown by `checkSecurity`.  The filter can throw an exception for the tile being missing instead of a security one.